R/salesforce.R

```diff
@@ -90,10 +95,58 @@ querySalesforceDataFromTable <- function(server = NULL, username, password, secu
   }
   query <- stringr::str_c("SELECT ", stringr::str_c(columns, collapse = ", "), " FROM ", table)
   if (!is.null(conditions)) {
-    query <- stringr::str_c(query, " ", conditions)
+    conditionLength = length(conditions)
+    conditionCount <- 0
+    whereClause <- ""
+    for(i in 1:conditionLength) {
+      # Check if the filter condition uses parameter (i.e. detect @{param})
+      hasParameter <- stringr::str_detect(conditions[i], "\\@\\{([^\\}]+)\\}")
+      # make sure to resolve parameter @{} syntax.
+      condition <- glue_exploratory(conditions[i], .transformer=salesforce_glue_transformer, .envir = parent.frame())
+      # make sure to resolve ${} syntax for Salesforce filter.
+      condition <- glue_salesforce(condition)
+      # When the IN (NULL) condition is detected after resolving parameter, remove the condition to support select "ALL" option.
+      # NOTE: IN (@{PARAM}) became IN (NULL) when nothing is selected from UI and Salesforce SOQL does not allow using IS_NULL function,
+      # so remove the "Column IN (NULL)" condition.
+      if (hasParameter && stringr::str_detect(condition, "IN \\(NULL\\)")) {
```

Comment:
Make it more strict by detecting the string only at the end of the condition.

Reply:
Added condition to check if it at the ends of condition.